4.analyze_data/notebooks/UMAP/nbconverted/2.find_sc.py

```diff
@@ -0,0 +1,217 @@
+    plate = random_row['Metadata_Plate'].values[0]
+    well = random_row['Metadata_Well'].values[0]
+    site = random_row['Metadata_Site'].values[0]
+
+    # Initialize a list to store file paths
+    file_paths = []
+
+    # Create 5 different file paths with "d0" through "d4" suffixes
+    for i in range(5):
+        filename = f"{images_dir}/{plate}_{well}{site}d{i}_illumcorrect.tiff"
+        file_paths.append(filename)
+
+    print("Randomly selected row:")
+    print(random_row)
+    print("Generated filenames:")
+    for path in file_paths:
+        print(path)
+
+    # Initialize empty lists to store the images for each channel
+    blue_channel = []
+    green_channel = []
+    red_channel = []
+
+    # Iterate over files in the directory
+    for file_path in file_paths:
+        filename = pathlib.Path(file_path).name
+        if 'd0' in filename:
+            blue_channel_image = cv2.imread(str(file_path), cv2.IMREAD_UNCHANGED)
+            blue_channel.append(blue_channel_image)
+        elif 'd1' in filename:
+            green_channel_image = cv2.imread(str(file_path), cv2.IMREAD_UNCHANGED)
+            green_channel.append(green_channel_image)
+        elif 'd4' in filename:
+            red_channel_image = cv2.imread(str(file_path), cv2.IMREAD_UNCHANGED)
+            red_channel.append(red_channel_image)
+
+    # Stack the images for each channel along the channel axis
+    blue_channel_stack = np.stack(blue_channel, axis=-1)
+    green_channel_stack = np.stack(green_channel, axis=-1)
+    red_channel_stack = np.stack(red_channel, axis=-1)
+
+    # Scale the pixel values to fit within the 16-bit range (0-65535)
+    blue_channel_stack = (blue_channel_stack / np.max(blue_channel_stack) * 65535).astype(np.uint16)
+    green_channel_stack = (green_channel_stack / np.max(green_channel_stack) * 65535).astype(np.uint16)
+    red_channel_stack = (red_channel_stack / np.max(red_channel_stack) * 65535).astype(np.uint16)
+
+    # Create the RGB image by merging the channels
+    composite_image = cv2.merge((red_channel_stack, green_channel_stack, blue_channel_stack)).astype(np.uint16)
+
+    # Path for saving comp images
+    comp_path = pathlib.Path(f"{comp_dir}/{info['composite_save_path']}")
+
+    # Save the composite 16-bit RGB tiff image
+    tf.imwrite(comp_path, composite_image)
+
+    # Load the composite image
+    composite_image = Image.open(comp_path)
+
+    # Assuming you have a DataFrame called "filtered_df" with center coordinates
+    center_x = random_row["Metadata_Cells_Location_Center_X"]
+    center_y = random_row["Metadata_Cells_Location_Center_Y"]
+
+    # Define the size of the cropping box (128x128 pixels)
+    box_size = 250
+
+    # Paths for saving cropped images
+    crop_path = pathlib.Path(f"{crop_dir}/{info['crop_save_path']}")
+
+    # Iterate through the center coordinates and crop each cell
+    for x, y in zip(center_x, center_y):
+        left = x - box_size // 2
```

Comment:
would it be easier to define a square 'radius' for defining the box sizes instead of floor dividing for each?
instead of a box size of `250` make a radius of `125`?

Reply:
I am not sure easier is the best way of putting it, I do believe that is another good way of performing this task. For me, the easiest thing is to set the pixel dimensions you want (e.g., 250 means 250x250, etc.) as it is straightforward what you want.

Honestly, this comes down to preference, but thanks for bringing up the different method!